arbnode/dataposter/data_poster.go

```diff
@@ -341,7 +341,7 @@ func (p *DataPoster) saveTx(ctx context.Context, prevTx, newTx *storage.QueuedTr
 }
 
 func (p *DataPoster) sendTx(ctx context.Context, prevTx *storage.QueuedTransaction, newTx *storage.QueuedTransaction) error {
-	if prevTx != newTx {
+	if prevTx == nil || (newTx != nil && newTx.FullTx.Hash() != prevTx.FullTx.Hash()) {
```

Comment:
Nit: we can assume newTx is not nil here to simplify this condition a bit, and in fact, we do `newTx.FullTx` a couple lines down which would panic on nil anyways.

Reply:
Done.